### DISCUSSION TITLE
what is the gko::remove_complex<ValueType>?

### DISCUSSION THREAD

**Discussion by ztdepztdep** [ginkgo-project/ginkgo#1003] [Q&A] [answered]
I have noticed that there is a data type gko::remove_complex<ValueType>, could you pleaes provide some infors about this type?

**Comment by upsj** [comment, marked as answer]
This is not a data type, but an [alias template](https://en.cppreference.com/w/cpp/language/type_alias) that essentially does two things:
* For value types like `double` and `std::complex<float>`, it turns complex types into their underlying type (i.e. `std::complex<double>` into `double`), and leaves other types like `double` unchanged.
* For template types like `Dense<ValueType>`, it turns `Dense<std::complex<double>>` into `Dense<double>`, i.e. applying `remove_complex` to the first template parameter.

But I have a general request for you: Please only use issues for things that are broken. If you have a question about Ginkgo, that's perfectly fine, but we use the Q&A discussion section for those. I already moved a few of your issues into the Q&A section.
